### ISSUE TITLE
Shape error when training with train_csv.py

### ISSUE THREAD

**Issue by Checksum** [fizyr/keras-retinanet#82]
Just installed from master and tried training on ~40k images. `training.csv` and `labels.csv` are in the expected format.

ML noob so pardon the stupid question: do the images have to be resized before the training is run? I'll dig into it, but putting it up here in case I'm doing something dumb.

Thanks!

```
> python examples\train_csv.py D:\training-cleaned.csv D:\labels.csv --gpu=0                                                                                   
Using TensorFlow backend.                                                                                                                                      

<snip>

None                                                                                                                                                           
Epoch 1/20                                                                                                                                                     
Traceback (most recent call last):                                                                                                                             
  File "examples\train_csv.py", line 117, in <module>                                                                                                          
    keras.callbacks.ReduceLROnPlateau(monitor='loss', factor=0.1, patience=2, verbose=1, mode='auto', epsilon=0.0001, cooldown=0, min_lr=0),                   
  File "C:\Users\srinath\.virtualenv\tensorflow-gpu\lib\site-packages\keras\legacy\interfaces.py", line 87, in wrapper                                         
    return func(*args, **kwargs)                                                                                                                               
  File "C:\Users\srinath\.virtualenv\tensorflow-gpu\lib\site-packages\keras\engine\training.py", line 2114, in fit_generator                                   
    class_weight=class_weight)                                                                                                                                 
  File "C:\Users\srinath\.virtualenv\tensorflow-gpu\lib\site-packages\keras\engine\training.py", line 1826, in train_on_batch                                  
    check_batch_axis=True)                                                                                                                                     
  File "C:\Users\srinath\.virtualenv\tensorflow-gpu\lib\site-packages\keras\engine\training.py", line 1411, in _standardize_user_data                          
    exception_prefix='target')                                                                                                                                 
  File "C:\Users\srinath\.virtualenv\tensorflow-gpu\lib\site-packages\keras\engine\training.py", line 153, in _standardize_input_data                          
    str(array.shape))                                                                                                                                          
ValueError: Error when checking target: expected regression to have shape (None, None, 4) but got array with shape (1, 103113, 5)                              
```

**Comment by Checksum** [comment]
As expected, I had the wrong version of Keras. Guess https://github.com/fizyr/keras-retinanet/pull/75 will help with that.

**Comment by yhenon** [comment]
This is a known issue (see #66). A fix to this issue has been submitted to the keras repo and merge. Until the next version is released, downgrade keras to 2.0.9
